process/txcost/gasUsed.go

```diff
@@ -0,0 +1,40 @@
+package txcost
+
+import (
+	"runtime/debug"
+
+	"github.com/ElrondNetwork/elrond-proxy-go/data"
+)
+
+func (tcp *transactionCostProcessor) prepareGasUsed(senderShardID, receiverShardID uint32, res *data.TxCostResponseData) {
+	numResponses := len(tcp.responses)
+	extra := 0
+	if senderShardID != receiverShardID {
+		extra = 1
+	}
+
+	syncChan := make(chan struct{})
+	go func(c chan struct{}) {
```

Comment:
could have omitted the go routine part and the waiting channel

Reply:
refactored.